object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -279,6 +304,253 @@ private static String getExactStringValue(Tree tree, ValueAnnotatedTypeFactory f
     return null;
   }
 
+  @Override
+  public void postAnalyze(ControlFlowGraph cfg) {
+    checkAlwaysCall(cfg);
+    super.postAnalyze(cfg);
+  }
+
+  private void checkAlwaysCall(ControlFlowGraph cfg) {
+
+    Set<Pair<BlockImpl, Set<Tree>>> visited = new HashSet<>();
+    visited.add(Pair.of((BlockImpl) cfg.getEntryBlock(), Collections.emptySet()));
+
+    Deque<Pair<BlockImpl, Set<Pair<LocalVariableNode, Tree>>>> worklist = new ArrayDeque<>();
+    worklist.add(Pair.of((BlockImpl) cfg.getEntryBlock(), Collections.emptySet()));
+
+    while (!worklist.isEmpty()) {
+
+      Pair<BlockImpl, Set<Pair<LocalVariableNode, Tree>>> state = worklist.removeLast();
+      List<Node> nodes = getBlockNodes(state.first);
+      Set<Pair<LocalVariableNode, Tree>> newDefs = new HashSet<>(state.second);
+
+      for (Node node : nodes) {
+        if (node instanceof AssignmentNode) {
+          Node lhs = ((AssignmentNode) node).getTarget();
+          Node rhs = ((AssignmentNode) node).getExpression();
+
+          if (lhs instanceof LocalVariableNode && hasAlwaysCall(lhs.getType())) {
+
+            // Reassignment to the lhs
+            if (isVarInDefs(newDefs, (LocalVariableNode) lhs)) {
+              Pair<LocalVariableNode, Tree> latestAssignmentPair =
+                  getAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
+              reportAlwaysCallErrors(latestAssignmentPair, getStoreBefore(node), null);
+              newDefs.remove(Pair.of((LocalVariableNode) lhs, latestAssignmentPair));
+            }
+
+            // If the rhs is ObjectCreationNode, MethodInvocationNode, or
+            // LocalVariableNode that exists in the newDefs (Note that if the localVariableNode
+            // exists in the newDefs it means it is assigned to a non-null values), then it adds
+            // the the AssignmentNode to the newDefs.
+            if ((rhs instanceof ObjectCreationNode)
+                || (rhs instanceof MethodInvocationNode)
+                || (rhs instanceof LocalVariableNode
```

Comment:
Can we remove this condition here, and fully handle ownership transfer in the condition below (addition and deletion from newDefs)?

Reply:
Yes, we can! Thanks!